### PR TITLE
clarify docs on validation and strict type checking

### PR DESCRIPTION
**Pull request by paxcodes**
## Change Summary

Clarify docs about Pydantic's support for custom validation and strict type checking, despite that pydantic is primarily a parsing library.

Content is based on comments by @cknoll [[link](https://github.com/samuelcolvin/pydantic/issues/578#issuecomment-711144799)] and @PyAntony [[link](https://github.com/samuelcolvin/pydantic/issues/578#issuecomment-706621721)) <sup>[1](#myfootnote1)</sup>

## Related issue number

#578

## Checklist

* [n/a] Unit tests for the changes exist
* [n/a] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

---

<sub><a name="myfootnote1">1</a>: I'd need your email / GitHub-provided no-reply email if you'd like GitHub to recognize you as co-authors of the commit.</span></sub>

### CODE REVIEW

**Comment by paxcodes** [comment]
Please review @PrettyWood

File: changes/2855-paxcodes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Clarify documentation about `pydantic`'s support for custom validation and strict type checking, despite `pydantic` being primarily a parsing library.
```
Comment (PrettyWood) [inline changes/2855-paxcodes.md:1]:
```suggestion
Clarify documentation about _pydantic_'s support for custom validation and strict type checking, despite _pydantic_ being primarily a parsing library.
```

File: docs/usage/models.md [hunk 2]
```diff
@@ -613,6 +615,8 @@ _(This script is complete, it should run "as is")_
 This is a deliberate decision of *pydantic*, and in general it's the most useful approach. See 
 [here](https://github.com/samuelcolvin/pydantic/issues/578) for a longer discussion on the subject.
 
+Nevertheless, [strict type checking](types.md#strict-types) is supported.
```
Comment (PrettyWood) [inline docs/usage/models.md:618]: I would have said partially supported. We'll need to wait for https://github.com/samuelcolvin/pydantic/issues/1098 to have a full (and easy to use) strict type checking

**Review by PrettyWood — CHANGES_REQUESTED** [review]

**Comment by PrettyWood** [comment]
Please update and rebase/merge with master. Thanks!

**Comment by paxcodes** [comment]
Please review.

**Review by PrettyWood — APPROVED** [review]
Great!

**Pull request merged**